### ISSUE TITLE
The project system has encountered an error. while opening up project properties page.

### ISSUE THREAD

**Issue by MarkMichaelis** [dotnet/roslyn-project-system#1223]
---------------------------
Microsoft Visual Studio
While opening up the project properties page on the Coelesce.Domain project within the https://github.com/IntelliTect/Coalesce/tree/VS2017RC3.15.0.26116.0.D15REL branch (https://github.com/IntelliTect/Coalesce/commit/4358d4821cdf98c6f323ee39ae62669a48124aca) the following error dialog appeared:
`
---------------------------
The project system has encountered an error.
An internal error occurred. Please contact Microsoft Product Support Services.
A diagnostic log has been written to the following location: "[...\VsProjectFault_ca41dfbc-1faf-4aa6-be3c-c3cfb3baad83.failure.txt](https://www.dropbox.com/s/k0tbk22fbizyqxy/VsProjectFault_ca41dfbc-1faf-4aa6-be3c-c3cfb3baad83.failure.txt?dl=1)".
---------------------------
OK   
---------------------------
`

The log file is found at: https://www.dropbox.com/s/k0tbk22fbizyqxy/VsProjectFault_ca41dfbc-1faf-4aa6-be3c-c3cfb3baad83.failure.txt?dl=1

/cc @srivatsn @morrisjoe

**Comment by mavasani** [comment]
I see DTAR on the callstack, tagging @davkean 

```
=====================
1/17/2017 1:39:38 PM
Crippling
System.AggregateException: One or more errors occurred. ---> Microsoft.Assumes+InternalErrorException: An internal error occurred. Please contact Microsoft Product Support Services.
   at Microsoft.Assumes.Fail(String message, Boolean showAssert)
   at Microsoft.VisualStudio.ProjectSystem.VS.References.DesignTimeAssemblyResolution.GetAllResolvedReferences()
   at Microsoft.VisualStudio.ProjectSystem.VS.References.DesignTimeAssemblyResolution.ResolveReferences(String[] originalNames, AssemblyName[] assemblyName, VsResolvedAssemblyPath[] assemblyPaths)
   at Microsoft.VisualStudio.ProjectSystem.VS.References.DesignTimeAssemblyResolution.ResolveAssemblyPathInTargetFx(String[] prgAssemblySpecs, UInt32 cAssembliesToResolve, VsResolvedAssemblyPath[] prgResolvedAssemblyPaths, UInt32& pcResolvedAssemblyPaths)
   at Microsoft.VisualStudio.Shell.Design.VsTargetFrameworkUniverse.ResolveAssemblyPath(String assemblySpec)
   at Microsoft.VisualStudio.Shell.Design.VsTargetFrameworkUniverse.ResolveAssembly(AssemblyName assemblyName)
   at Microsoft.VisualStudio.Shell.Design.VsTargetFrameworkUniverse.GetSystemAssembly()
   at Microsoft.VisualStudio.Shell.Design.VsTargetFrameworkUniverse..ctor(IDesignTimeAssemblyLoader assemblyLoader, IVsDesignTimeAssemblyResolution projectAssemblyResolution, IVsSmartOpenScope dispenser)
   at Microsoft.VisualStudio.Shell.Design.VsTargetFrameworkUniverse.GetUniverse(IDesignTimeAssemblyLoader assemblyLoader, IVsDesignTimeAssemblyResolution projectAssemblyResolution, IVsSmartOpenScope dispenser)
   at Microsoft.VisualStudio.Shell.Design.VsTargetFrameworkProvider..ctor(IVsDesignTimeAssemblyResolution assemblyResolution, IDesignTimeAssemblyLoader assemblyLoader, TypeDescriptionProvider parentProvider, IVsSmartOpenScope openScope)
   at Microsoft.VisualStudio.ProjectUtilities.GetTargetFrameworkProvider(IServiceProvider provider, IDesignTimeAssemblyLoader assemblyLoader, IVsHierarchy hierarchy, Object instance)
   at Microsoft.VisualStudio.Design.VSTypeResolutionService.get_TypeProvider()
   at Microsoft.VisualStudio.Design.VSTypeResolutionService.ClearReferenceEntries(Boolean normal, Boolean project, Boolean disposing)
   at Microsoft.VisualStudio.Design.VSTypeResolutionService.VSLangProj._dispReferencesEvents.ReferenceRemoved(Reference r)
   at Microsoft.VisualStudio.ProjectSystem.VS.Implementation.Package.Automation.OAVSReferences.OnReferenceRemoved(Reference reference)
   at Microsoft.VisualStudio.ProjectSystem.VS.Implementation.Package.ReferencesHostBridge.ApplyAsync(ReferencesDelta value)
   at Microsoft.VisualStudio.ProjectSystem.VS.Implementation.Package.ProjectHostBridge`3.<>c__DisplayClass36_1.<<InitializeCoreAsync>b__3>d.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
```

**Comment by davkean** [comment]
Thanks for the report - I've submitted a fix.

**Comment by davkean** [comment]
This was fixed with #1239.
